src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -228,28 +254,71 @@ public Path exportProgrammingExerciseRepositories(ProgrammingExercise exercise,
      * @return a zipped file
      */
     public Optional<File> exportInstructorRepositoryForExercise(long exerciseId, RepositoryType repositoryType, Path outputDir, List<String> exportErrors) {
+        var exerciseOrEmpty = loadExerciseForRepoExport(exerciseId, repositoryType.getName(), exportErrors);
+        if (exerciseOrEmpty.isEmpty()) {
+            return Optional.empty();
+        }
+
+        var exercise = exerciseOrEmpty.get();
+
+        String zippedRepoName = getZippedRepoName(exercise, repositoryType.getName());
+
+        var repositoryUrl = exercise.getRepositoryURL(repositoryType);
+
+        return exportRepository(repositoryUrl, repositoryType.getName(), zippedRepoName, exercise, outputDir, exportErrors);
```

Comment:
I would recommend removing some empty lines here, occasional usage makes each one more significant/helpful for understanding the code. Same for the next method.